From cd50514efe157aef06b76dcf83fc1fd326e41d4c Mon Sep 17 00:00:00 2001
From: Harry Yoo <harry.yoo@oracle.com>
Date: Mon, 16 Mar 2026 15:19:49 +0900
Subject: [PATCH 68/87] WARNING in folio_add_new_anon_rmap

#syz test
---
 mm/userfaultfd.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/userfaultfd.c b/mm/userfaultfd.c
index 83a9377130fd..1d1a226f1126 100644
--- a/mm/userfaultfd.c
+++ b/mm/userfaultfd.c
@@ -824,7 +824,7 @@ static __always_inline ssize_t mfill_atomic(struct userfaultfd_ctx *ctx,
 		.dst_start = dst_start,
 		.src_start = src_start,
 		.flags = flags,
-
+		.len = len,
 		.src_addr = src_start,
 		.dst_addr = dst_start,
 	};
-- 
2.50.1

