From abdd00eb33d598c6fbfdb9bfbbe2882075f00eae Mon Sep 17 00:00:00 2001
From: Nikita Kalyazin <kalyazin@amazon.com>
Date: Mon, 2 Mar 2026 15:13:17 +0000
Subject: [PATCH 81/94] fixup for uffd: export mm symbols

Signed-off-by: Nikita Kalyazin <kalyazin@amazon.com>
---
 mm/mempolicy.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mm/mempolicy.c b/mm/mempolicy.c
index 3d797d47a040..1dbbbb28a36e 100644
--- a/mm/mempolicy.c
+++ b/mm/mempolicy.c
@@ -2442,6 +2442,7 @@ struct folio *folio_alloc_mpol_noprof(gfp_t gfp, unsigned int order,
 	set_page_refcounted(page);
 	return page_rmappable_folio(page);
 }
+EXPORT_SYMBOL_FOR_MODULES(folio_alloc_mpol_noprof, "kvm");
 
 /**
  * vma_alloc_folio - Allocate a folio for a VMA.
-- 
2.50.1

