From c7ee86ceb3a0cdfe1532099ec7f12f069381db33 Mon Sep 17 00:00:00 2001
From: Nikita Kalyazin <kalyazin@amazon.com>
Date: Mon, 2 Mar 2026 15:13:17 +0000
Subject: [PATCH 81/84] fixup for uffd: export mm symbols

Signed-off-by: Nikita Kalyazin <kalyazin@amazon.com>
---
 mm/filemap.c   | 1 +
 mm/mempolicy.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/mm/filemap.c b/mm/filemap.c
index 3e4579e4b8bb..041c7719e524 100644
--- a/mm/filemap.c
+++ b/mm/filemap.c
@@ -261,6 +261,7 @@ void filemap_remove_folio(struct folio *folio)
 
 	filemap_free_folio(mapping, folio);
 }
+EXPORT_SYMBOL_FOR_MODULES(filemap_remove_folio, "kvm");
 
 /*
  * page_cache_delete_batch - delete several folios from page cache
diff --git a/mm/mempolicy.c b/mm/mempolicy.c
index 3d797d47a040..1dbbbb28a36e 100644
--- a/mm/mempolicy.c
+++ b/mm/mempolicy.c
@@ -2442,6 +2442,7 @@ struct folio *folio_alloc_mpol_noprof(gfp_t gfp, unsigned int order,
 	set_page_refcounted(page);
 	return page_rmappable_folio(page);
 }
+EXPORT_SYMBOL_FOR_MODULES(folio_alloc_mpol_noprof, "kvm");
 
 /**
  * vma_alloc_folio - Allocate a folio for a VMA.
-- 
2.50.1

