From 43206a265ef1ad72604e7940e8f72a11244d86ea Mon Sep 17 00:00:00 2001
From: Nikita Kalyazin <kalyazin@amazon.com>
Date: Mon, 26 Jan 2026 12:37:52 +0000
Subject: [PATCH 83/94] fixup for uffd + dmr

Signed-off-by: Nikita Kalyazin <kalyazin@amazon.com>
---
 virt/kvm/guest_memfd.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/virt/kvm/guest_memfd.c b/virt/kvm/guest_memfd.c
index eda34a3e6a98..7bd90e91c3bd 100644
--- a/virt/kvm/guest_memfd.c
+++ b/virt/kvm/guest_memfd.c
@@ -621,6 +621,12 @@ static int kvm_gmem_filemap_add(struct folio *folio,
 		return err;
 	}
 
+	err = kvm_gmem_folio_zap_direct_map(folio);
+	if (err) {
+		folio_unlock(folio);
+		return err;
+	}
+
 	return 0;
 }
 
-- 
2.50.1

