From e68cb787322339cc697b751eb373da08a64c72cf Mon Sep 17 00:00:00 2001
From: Nikita Kalyazin <kalyazin@amazon.com>
Date: Mon, 26 Jan 2026 12:37:52 +0000
Subject: [PATCH 83/84] fixup for uffd + dmr

Signed-off-by: Nikita Kalyazin <kalyazin@amazon.com>
---
 virt/kvm/guest_memfd.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/virt/kvm/guest_memfd.c b/virt/kvm/guest_memfd.c
index 4cffebd312e8..157f0a5c0f93 100644
--- a/virt/kvm/guest_memfd.c
+++ b/virt/kvm/guest_memfd.c
@@ -659,6 +659,12 @@ static int kvm_gmem_filemap_add(struct folio *folio,
 		return err;
 	}
 
+	err = kvm_gmem_folio_zap_direct_map(folio);
+	if (err) {
+		folio_unlock(folio);
+		return err;
+	}
+
 	return 0;
 }
 
-- 
2.50.1

