From 392d147b89e82cfb5a8162a0a097d8a245c0fa60 Mon Sep 17 00:00:00 2001
From: Nikita Kalyazin <kalyazin@amazon.com>
Date: Mon, 26 Jan 2026 12:37:52 +0000
Subject: [PATCH 84/87] fixup for uffd + dmr

Signed-off-by: Nikita Kalyazin <kalyazin@amazon.com>
---
 virt/kvm/guest_memfd.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/virt/kvm/guest_memfd.c b/virt/kvm/guest_memfd.c
index 336c23e33ff0..86c17ea6b38f 100644
--- a/virt/kvm/guest_memfd.c
+++ b/virt/kvm/guest_memfd.c
@@ -627,6 +627,12 @@ static int kvm_gmem_filemap_add(struct folio *folio,
 		return err;
 	}
 
+	err = kvm_gmem_folio_zap_direct_map(folio);
+	if (err) {
+		folio_unlock(folio);
+		return err;
+	}
+
 	return 0;
 }
 
-- 
2.50.1

