From 6e2089467372ffc9d61b0902fd237a7cdd1c6660 Mon Sep 17 00:00:00 2001
From: Andreea Florescu <fandree@amazon.com>
Date: Tue, 27 Feb 2018 20:25:03 +0200
Subject: [PATCH] vhost: fix alignment for vhost_memory

Signed-off-by: Andreea Florescu <fandree@amazon.com>
---
 vhost_gen/src/vhost.rs | 1 +
 1 file changed, 1 insertion(+)

diff --git a/vhost_gen/src/vhost.rs b/vhost_gen/src/vhost.rs
index 7d1be6b..c6f3fdc 100644
--- a/vhost_gen/src/vhost.rs
+++ b/vhost_gen/src/vhost.rs
@@ -500,6 +500,7 @@ pub struct vhost_memory {
     pub nregions: __u32,
     pub padding: __u32,
     pub regions: __IncompleteArrayField<vhost_memory_region>,
+    __force_alignment: [u64; 0],
 }
 #[test]
 fn bindgen_test_layout_vhost_memory() {
-- 
2.7.4

